Objects/unionobject.c

```diff
@@ -242,7 +303,36 @@ _Py_union_type_or(PyObject* self, PyObject* other)
         Py_RETURN_NOTIMPLEMENTED;
     }
 
-    PyObject *tuple = PyTuple_Pack(2, self, other);
+    if (self == Py_None) {
+        self = (PyObject *)&_PyNone_Type;
+    }
+    if (other == Py_None) {
+        other = (PyObject *)&_PyNone_Type;
```

Comment:
Do we need to decref `Py_None` here?

Reply:
Nope, it will lead to a python crash on finalization.